front/src/modules/settings/data-model/object-details/components/SettingsObjectFieldDataType.tsx

```diff
@@ -24,21 +40,33 @@ const StyledDataType = styled.div<{ value: FieldMetadataType }>`
       : ''}
 `;
 
-type SettingsObjectFieldDataTypeProps = {
-  value: FieldMetadataType;
-};
+const StyledLabelContainer = styled.div`
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+`;
 
 export const SettingsObjectFieldDataType = ({
+  onClick,
   value,
+  Icon = dataTypes?.[value]?.Icon,
+  label = dataTypes?.[value]?.label,
 }: SettingsObjectFieldDataTypeProps) => {
   const theme = useTheme();
 
-  const { label, Icon } = dataTypes?.[value];
-
   return (
-    <StyledDataType value={value}>
-      <Icon size={theme.icon.size.sm} />
-      {label}
+    <StyledDataType onClick={onClick} value={value}>
+      <ClassNames>
```

Comment:
this is introducing a new pattern, we should discuss it :)
Is there another nice way to achieve the same goal?

Reply:
As discussed: I've wrapped the Icon with `styled` instead.